Textile/Textile.sublime-syntax

```diff
@@ -1,225 +1,1111 @@
+        ({{textile_alignment_optional}})
+        ({{textile_attributes_optional_class_style}})
+        (\.\.)
+        [ ](?=\S)
+      scope: entity.name.section.paragraph.textile
       captures:
-        1: entity.name.type.textile
+        1: entity.name.tag.paragraph.textile
+        2: constant.other.indentation.textile
+        3: constant.other.alignment.textile
+        4: entity.name.type.textile
+        5: entity.name.tag.paragraph.textile
+      push: textile-extended-paragraph-body
     - match: |-
         (?x)
-        			"								# Start name, etc
-        				(?:							# Attributes
-        					# I swear, this is how the language is defined,
-        					# couldnt make it up if I tried.
-        					(?:\([^)]+\))?(?:\{[^}]+\})?(?:\[[^\]]+\])?
-        						# Class, Style, Lang
-        				  | (?:\{[^}]+\})?(?:\[[^\]]+\])?(?:\([^)]+\))?
-        						# Style, Lang, Class
-        				  | (?:\[[^\]]+\])?(?:\{[^}]+\})?(?:\([^)]+\))?
-        						# Lang, Style, Class
-        				)?
-        				([^"]+?)					# Link name
-        				\s?							# Optional whitespace
-        				(?:\(([^)]+?)\))?
-        			":								# End name
-        			(\w[-\w_]*)						# Linkref
-        			(?=[^\w\/;]*?(<|\s|$))			# Catch closing punctuation
-        											#  and end of meta.link
+        ^
+        (p)
+        ({{textile_indentation_optional}})
+        ({{textile_alignment_optional}})
+        ({{textile_attributes_optional_class_style}})
+        (\.)
+        [ ](?=\S)
+      scope: entity.name.section.paragraph.textile
+      captures:
+        1: entity.name.tag.paragraph.textile
+        2: constant.other.indentation.textile
+        3: constant.other.alignment.textile
+        4: entity.name.type.textile
+        5: entity.name.tag.paragraph.textile
+      push: textile-paragraph-body
+    - match: ^(?=\S)
+      push: textile-paragraph-body
+
+  textile-extended-paragraph-body:
+    - meta_scope: meta.paragraph.textile
+    - include: textile-next-block-pop
+    - include: html
+    - include: textile-inlines
+
+  textile-paragraph-body:
+    - meta_scope: meta.paragraph.textile
+    - include: textile-empty-line-pop
+    - include: html
+    - include: textile-inlines
+
+###[ BLOCKS / LISTS ]##########################################################
+
+  # https://textile-lang.com/doc/bulleted-unordered-lists
+  # https://textile-lang.com/doc/numbered-ordered-lists
+  # https://textile-lang.com/doc/definition-lists
+
+  textile-lists:
+    - match: '^(\*+)({{textile_attributes_optional}})'
+      captures:
+        1: markup.list.unnumbered.bullet.textile
+        2: markup.list.unnumbered.textile entity.name.type.textile
+      push: textile-unordered-list-item
+    - match: '^(#+)({{textile_attributes_optional}})'
+      captures:
+        1: markup.list.numbered.bullet.textile
+        2: markup.list.numbered.textile entity.name.type.textile
+      push: textile-ordered-list-item
+    - match: '^[-]'
```

Comment:
That's micro-tuning, but the following change reduces compiled syntax cache by about 2-3kB.

```suggestion
    - match: '^-'
```